proto/cosmos/crypto/v1beta1/keys.proto

```diff
@@ -0,0 +1,32 @@
+syntax = "proto3";
+package cosmos.crypto.v1beta1;
+
+import "gogoproto/gogo.proto";
+import "google/protobuf/any.proto";
+
+option go_package = "github.com/cosmos/cosmos-sdk/crypto/keys";
+
+// MultisigThresholdPubKey specifies a public key type which nests multiple public
+// keys and a threshold
+message MultisigThresholdPubKey {
+    option (gogoproto.goproto_getters) = false;
+
+    uint32 threshold = 1 [(gogoproto.customname) = "K", (gogoproto.moretags) = "yaml:\"threshold\""];
+    repeated google.protobuf.Any public_keys = 2 [(gogoproto.customname) = "PubKeys", (gogoproto.moretags) = "yaml:\"pubkeys\""];
+}
+
+// LegacyAminoMultisigThresholdPubKey is a version of MultisigThresholdPubKey which uses legacy amino address rules
+message LegacyAminoMultisigThresholdPubKey {
+    uint32 threshold = 1 [(gogoproto.customname) = "K", (gogoproto.moretags) = "yaml:\"threshold\""];
+    repeated google.protobuf.Any public_keys = 2 [(gogoproto.customname) = "PubKeys", (gogoproto.moretags) = "yaml:\"pubkeys\""];
+}
+
+// Secp256K1PubKey defines a secp256k1 public key
+message Secp256K1PubKey {
+    bytes key = 1 [(gogoproto.casttype) = "github.com/cosmos/cosmos-sdk/crypto/keys/secp256k1.PubKey"];
+}
+
+// Secp256K1PrivKey defines a secp256k1 private key
+message Secp256K1PrivKey {
+    bytes key = 1 [(gogoproto.casttype) = "github.com/cosmos/cosmos-sdk/crypto/keys/secp256k1.PrivKey"];
+}
```

Comment:
One question I have i whether we should create `{Ed,Sr}25519{Pub,Priv}Key` message, which casttype to TM's types.

From my understanding, if we want to use Any for a public key scheme, we need to create proto types for it.